src/js/controllers/tab-scan.js

```diff
@@ -57,68 +68,133 @@ angular.module('copayApp.controllers').controller('tabScanController', function(
 
 
   $scope.$on("$ionicView.beforeEnter", function(event, data) {
-    console.log('tab-scan $ionicView.beforeEnter');
-    console.log('data', data.stateParams.returnRoute);
-    _handleCapabilities();
-    $timeout(() => {
-        _refreshScanView()
-        if (isSafari() !== "SAFARI") {
-          $scope.cameraIssue = true
-        }
-      },
-    5000);
-    $scope.returnRoute = data.stateParams.returnRoute || false;
+    if (platformInfo.cameraType === 'native') {
+      _handleCapabilities();
+      $timeout(function() {
+          _refreshScanView()
+        },
+        5000);
+    }
+    $scope.returnRoute = $state.params.returnRoute || false;
   });
 
   $scope.$on("$ionicView.afterEnter", function() {
     // try initializing and refreshing status any time the view is entered
-    if(!scannerService.isInitialized()){
+    if(!scannerService.isInitialized() && platformInfo.cameraType === 'native'){
       scannerService.gentleInitialize();
     }
     activate();
   });
 
-  function isSafari() {
-    var ua = navigator.userAgent.toLowerCase();
-    if (ua.indexOf('safari') != -1) {
-      if (ua.indexOf('chrome') > -1) {
-        return "CHROME";
-      } else {
-        return "SAFARI";
-      }
+  $scope.$on("$ionicView.afterLeave", function() {
+    scannerService.deactivate();
+  });
+
+
+  function stopWebRtcCamera() {
+    clearInterval($scope.videoScanInterval)
+    var video = document.getElementById('webRtcScanner')
+    if ($scope.stream) {
+      $scope.stream.getVideoTracks().forEach(function(track) {
+        track.stop()
+      } )
     }
-    return "OTHER";
   }
 
```

Comment:
Web RTC is for the PWA and the other camera func is for in browser use?

Reply:
WebRTC is for anything that supports it. Camera is a backup when webRTC isn't fully supported. (iOS 11.3 PWA)